src/main/java/com/google/cloud/teleport/io/WindowedFilenamePolicy.java

```diff
@@ -137,22 +156,37 @@ public ResourceId unwindowedFilename(
   /**
    * Resolves any date variables which exist in the output directory path. This allows for the
    * dynamically changing of the output location based on the window end time.
+   * A custom pattern can be used, if provided.
    *
    * @return The new output directory with all variables resolved.
    */
   private ResourceId resolveWithDateTemplates(
       ValueProvider<String> outputDirectoryStr, BoundedWindow window) {
     ResourceId outputDirectory = FileSystems.matchNewResource(outputDirectoryStr.get(), true);
-
+  
     if (window instanceof IntervalWindow) {
       IntervalWindow intervalWindow = (IntervalWindow) window;
       DateTime time = intervalWindow.end().toDateTime();
       String outputPath = outputDirectory.toString();
-      outputPath = outputPath.replace("YYYY", YEAR.print(time));
-      outputPath = outputPath.replace("MM", MONTH.print(time));
-      outputPath = outputPath.replace("DD", DAY.print(time));
-      outputPath = outputPath.replace("HH", HOUR.print(time));
-      outputPath = outputPath.replace("mm", MINUTE.print(time));
+      String customDateTimePattern = pattern.get();
+    
+      if (customDateTimePattern.equals("")) {
+        outputPath = outputPath.replace("YYYY", YEAR.print(time));
+        outputPath = outputPath.replace("MM", MONTH.print(time));
+        outputPath = outputPath.replace("DD", DAY.print(time));
+        outputPath = outputPath.replace("HH", HOUR.print(time));
+        outputPath = outputPath.replace("mm", MINUTE.print(time));
```

Comment:
I think what we need is a way for the user to override what the pattern would be for each of the 5 components (year, month, day, hour and minute).

For example, a user may specify that the replacement pattern for minute should be `mx` instead of the default `mm`.

Reply:
What should the input format for this option be? A comma separated String of the overridden values? Something like `mm=mx,HH=hx`...

Reply:
The fluent builder pattern can have methods that allow us to override the patterns for each of these. So something like this:

```
WindowedFilenamePolicy policy = WindowedFilenamePolicy.of(<existing_input_params>);
...
// Override minute pattern from default "mm" to "xx"
policy.setMinutePattern("xx");

// Override day pattern from default "DD" to "ZZ"
policy.setDayPattern("ZZ");

```

Reply:
So does this mean each template that uses the `WindowedFilenamePolicy` class would have to have a separate input parameter for each Date component? Like:
```
WindowedFilenamePolicy.newBuilder()
                      .withOutputDirectory(options.getOutputDirectory())
                      .withOutputFilenamePrefix(options.getOutputFilenamePrefix())
                      .withShardTemplate(options.getOutputShardTemplate())
                      .withSuffix(options.getOutputFilenameSuffix())
                      .withYearPattern(options.getYearPattern())
                      .withMonthPattern(options.getMonthPattern()) // and so on
                      .build();
```  
\
Currently, I've changed the class to use the Builder pattern as follows: 
(assuming we're taking all the patterns to be overridden in one input parameter `--customDateTimePattern` from the user)

```
WindowedFilenamePolicy.newBuilder()
                      .withOutputDirectory(options.getOutputDirectory())
                      .withOutputFilenamePrefix(options.getOutputFilenamePrefix())
                      .withShardTemplate(options.getOutputShardTemplate())
                      .withSuffix(options.getOutputFilenameSuffix())
                      .withPattern(options.getCustomDateTimePattern())
                      .build();
```

What should the format for `options.getCustomDateTimePattern()` be?